datasetinsights/constants.py

```diff
@@ -53,3 +53,6 @@
     "allow_extra_args": True,
 }
 DEFAULT_DATASET_VERSION = "latest"
+
+SUMMARY_WRITER_MAX_QUEUE = 100
+SUMMARY_WRITER_FLUSH_SECS = 30
```

Comment:
Do you think this is the root cause of the previous error?

Reply:
I cant say for sure if the parameters are the root cause, but i was not able to replicate the error with the new parameters.

Reply:
Out of curiosity, do you still encounter this error without changing these default parameter? I am very interested to know the root cause of this issue and it's still not clear to me why changing these parameter fixed the issue.

Reply:
I will run one pipeline again with default SummaryWriter values

Reply:
I think we need to look into the tensorboardx source code to find the roor cause

Reply:
My hypothesis is that the SummaryWriter is trying to flush to disk and upload to gcs, but before the upload has finished it tries to upload another file causing the previous upload close its connection. Therefore increasing the max_queue size is the most important if there are alot of writes happening in fast after each other.

Reply:
Its based on this callback stack
```

textPayload: "Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.7/dist-packages/tensorboardX/event_file_writer.py", line 220, in run
    self._record_writer.flush()
  File "/usr/local/lib/python3.7/dist-packages/tensorboardX/event_file_writer.py", line 69, in flush
    self._py_recordio_writer.flush()
  File "/usr/local/lib/python3.7/dist-packages/tensorboardX/record_writer.py", line 189, in flush
    self._writer.flush()
  File "/usr/local/lib/python3.7/dist-packages/tensorboardX/record_writer.py", line 151, in flush
    self.blob.upload_from_string(upload_buffer.getvalue())
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/storage/blob.py", line 1961, in upload_from_string
    if_metageneration_not_match=if_metageneration_not_match,
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/storage/blob.py", line 1795, in upload_from_file
    if_metageneration_not_match,
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/storage/blob.py", line 1654, in _do_upload
    if_metageneration_not_match,
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/storage/blob.py", line 1321, in _do_multipart_upload
    response = upload.transmit(transport, data, object_metadata, content_type)
  File "/usr/local/lib/python3.7/dist-packages/google/resumable_media/requests/upload.py", line 106, in transmit
    retry_strategy=self._retry_strategy,
  File "/usr/local/lib/python3.7/dist-packages/google/resumable_media/requests/_helpers.py", line 136, in http_request
    return _helpers.wait_and_retry(func, RequestsMixin._get_status_code, retry_strategy)
  File "/usr/local/lib/python3.7/dist-packages/google/resumable_media/_helpers.py", line 150, in wait_and_retry
    response = func()
  File "/usr/local/lib/python3.7/dist-packages/google/auth/transport/requests.py", line 450, in request
    **kwargs
  File "/usr/local/lib/python3.7/dist-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/dist-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BrokenPipeError(32, 'Broken pipe'))
" 
```

Reply:
Its also possible that its was a unstable connection

Reply:
After testing writing with the SummaryWriter locally i don't think the above hypothesis is the root cause.

Reply:
I also think it's because of the connection since I didn't run into the issue with default settings locally.

Reply:
@AdamPalmarUnity @BlairLee 

So, the question still remains. Will users ran into this issue using Kubeflow pipeline? If yes, would the proposed change to increase queue size fix it? If no, we can close this issue.

Reply:
Clone two runs on kubeflow. If they have no issues, I think we are good to go.